### PR TITLE
Proxy rename support

### PR DESCRIPTION
**Pull request by omertuc**
[MGMT-16796](https://issues.redhat.com//browse/MGMT-16796)

See commit message

### CODE REVIEW

File: src/ocp_postprocess/proxy_rename.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    etcd_rename::fix_openshiftapiserver(etcd_client, proxy)
+        .await
+        .context("fixing openshiftapiserver")?;
+
+    etcd_rename::fix_kubeapiserver(etcd_client, proxy)
+        .await
+        .context("fixing kubeapiserver")?;
+
+    etcd_rename::fix_kubecontrollermanager(etcd_client, proxy)
```
[the remark this answers is not in the document]
Comment (omertuc) [reply src/ocp_postprocess/proxy_rename.rs:79]: They're not pods

[the remark this answers is not in the document]
Comment (omertuc) [reply src/ocp_postprocess/proxy_rename.rs:79]: `kind: KubeControllerManager`

File: src/ocp_postprocess/proxy_rename.rs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    filesystem_rename::fix_pods_yaml(proxy, dir).await.context("renaming pod yaml")?;
+
+    Ok(())
+}
+
+async fn fix_file_resources(proxy: &Proxy, file: &Path) -> Result<()> {
+    filesystem_rename::rename_proxy_env_file(proxy, file)
+        .await
+        .context("fixing etcd static pods")?;
```
Comment (mresvanis) [inline src/ocp_postprocess/proxy_rename.rs:57]:
```suggestion
        .context("renaming proxy env file")?;
```

File: src/ocp_postprocess/proxy_rename/filesystem_rename.rs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                tokio::spawn(async move {
+                    async move {
+                        rename_proxy_env(&mcdorig_file_path, &proxy)
+                            .await
+                            .context("renaming proxy.env.mcdorig")?;
+
+                        anyhow::Ok(())
+                    }
+                    .await
+                    .context(format!("fixing kubeconfig {:?}", &file_path))
```
Comment (mresvanis) [inline src/ocp_postprocess/proxy_rename/filesystem_rename.rs:55]:
```suggestion
                    .context(format!("fixing proxy.env.mcdorig {:?}", &file_path))
```

**Pull request merged**